core/arch/arm/arm.mk

```diff
@@ -9,6 +9,16 @@ CFG_KERN_LINKER_FORMAT ?= elf32-littlearm
 CFG_KERN_LINKER_ARCH ?= arm
 endif
 
+ifeq ($(CFG_TA_FLOAT_SUPPORT),y)
+# Use hard-float for floating pointer support in user TAs instead of
+# soft-float
+CFG_WITH_VFP ?= y
+ifeq ($(CFG_ARM64_core),y)
+# AArch64 has no fallback to soft-float
+$(call force,CFG_WITH_VFP,y)
+endif
+endif
```

Comment:
I'm a bit confused about the configuration here. Are the following true?
- To support hard-float TAs as well as soft-float TAs I just need to set CFG_TA_FLOAT_SUPPORT=y at build time
- To support soft-float TAs but not hard-float on ARM32, I would set CFG_TA_FLOAT_SUPPORT=y and CFG_WITH_VFP=n (so I would lose crypto extensions in TEE core). On ARM64, it is not possible to support only soft-float TAs and reject hard-float TAs.

Practically, I suppose it is fine. But I have to ask: would it be possible/useful to define two separate options, such as CFG_TA_HARD_FLOAT_SUPPORT and CFG_TA_SOFT_FLOAT_SUPPORT? IMO it would be easier to document (self-explanatory in fact), and possibly it would allow more combinations (or not, if we don't want to change the code too much).

If we keep only CFG_TA_FLOAT_SUPPORT then I would suggest adding a comment in conf.mk (see below).